test/fx/test_lazy_recompile.py

```diff
@@ -0,0 +1,57 @@
+# Owner(s): ["oncall: fx"]
+
+from torch.testing._internal.common_utils import TestCase, run_tests
+from torch import fx
+import torch
+import torch._export
+
+class TestLazyRecompile(TestCase):
```

Comment:
1) Add a test where you call, mutate the graph, and call again, so multiple recompiles are needed.

2) Add a test that asserts `needs_recompile()` is False after the compile happens (and true before it)